rfqmsg/request_test.go

```diff
@@ -0,0 +1,70 @@
+package rfqmsg
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightninglabs/taproot-assets/asset"
+	"github.com/lightninglabs/taproot-assets/internal/test"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/stretchr/testify/require"
+)
+
+// TestRequestMsgDataEncodeDecode tests the encoding and decoding of a request
+// message.
+func TestRequestMsgDataEncodeDecode(t *testing.T) {
+	t.Parallel()
+
+	// Create a random ID.
+	randomIdBytes := test.RandBytes(32)
+	id := ID(randomIdBytes)
+
+	// Create a random asset ID.
+	randomAssetIdBytes := test.RandBytes(32)
+	assetId := asset.ID(randomAssetIdBytes)
+
+	testCases := []struct {
+		testName string
+
+		id            ID
+		assetId       *asset.ID
+		assetGroupKey *btcec.PublicKey
+		assetAmount   uint64
+		bidPrice      lnwire.MilliSatoshi
+	}{
+		{
+			testName:      "asset_group_key_nil",
```

Comment:
nit: I know this is wrong in some existing places, but basically the test names should just use spaces which are then turned into underlines by the test framework for the actual name (a bit easier to read for humans if it's spaces here IMO).